python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -241,6 +226,28 @@ def evaluate(
             df_group_metrics = df_grouped_stats.apply(
                 metrics_helper.summarize_grouped_stats, axis=1
             )
+
+            # Add power analysis to group metric dataframe
+            df_group_metrics = metrics_helper.join_stat_sig_signal(df_group_metrics, group_keys,
+                                                                   eval_dict[EvalConfigConstants.METRICS],
+                                                                   group_metrics_stat_sig)
+
+            # Writing stat sig. groupwise results per metric
+            for metric in eval_dict[EvalConfigConstants.METRICS]:
+                stat_sig_df = metrics_helper.generate_stat_sig_based_metrics(df_group_metrics, metric, group_keys, metrics_dict)
+                if logs_dir:
+                    self.file_io.write_df(
+                        stat_sig_df,
+                        outfile=os.path.join(
+                            logs_dir, "stat_sig_" + metric + '_' + RelevanceModelConstants.GROUP_METRICS_CSV_FILE),
```

Comment:
@mohazahran FYI - this file will be lost in the docker image. You need to also update our internal training repo to add this to the saved model bundle to access it later.